bundler/doc/development/DEBUGGING.md

```diff
@@ -0,0 +1,76 @@
+# Debugging
+
+Debugging Bundler can be challenging, don't be discouraged 🤗.
+
+Make sure you've followed the [development setup](SETUP.md) docs before trying to debug.
+
+## Debugging in tests
+
+Your best option is to print debug in the test suite. Put `puts` statements anywhere you want to see an object or variable and you'll see your `puts` in the console output.
+
+```ruby
+puts "@definition: #{@definition}"
+puts "specification.class.name: #{specification.class.name}"
+puts "spec.method(:to_checksum).source_location: #{spec.method(:to_checksum).source_location}"
+# etc
+```
+
+Most tests use Open3 to run Bundler in a sub process and capture the output into a string, which makes it impossible to use pry even if you can get it to load.
+
+## Interactive debugging locally
+
+When testing Bundler locally, you can use pry for interactive debugging.
+
+The easiest way to test locally is to set up a directory with a Gemfile and run your Bundler shell alias (see the [development setup](SETUP.md) docs for instructions on setting up the alias).
+
+We recommend putting this directory inside of `tmp` so that your local tests don't accidentally get committed.
+
+```bash
+cd tmp
+mkdir [name of directory for local testing] && cd [name of directory for local testing]
+dbundle init
+```
+
+And then you should have a Gemfile ready for you to edit as needed for your testing.
+
+### RubyGems test case
+By default, the initialized Gemfile's remote is `"https://rubygems.org"`. You can add any gem hosted on rubygems and then run your Bundler shell alias (`dbundle`) to test your code.
+
+```ruby
+# frozen_string_literal: true
+
+source "https://rubygems.org"
+
+gem 'tiny_css'
+```
+
+Put a `binding.pry` anywhere you want the debugger to break. Run your Bundler shell alias.
```

Comment:
Because `binding.irb` is a REPL and not a debugger 🤷‍♀️ It's nice, but you can't really interactive debug. I can add that to these docs as a tool to use though

Reply:
That's it. Some info is at https://github.com/rubygems/rubygems/blob/030e285881ac99608fe8a23121b80ce1ce14ed6e/bundler/doc/development/SETUP.md#debugging-with-pry. Instead of this "complex" setup, you can do just `binding.irb` and you have basic debug console. For debug console, you can use `binding.break` (you need to `require 'debug'` somehow from [debug](https://github.com/ruby/debug) gem, which is distributed with Ruby if I remember well and there is no need to install additional dependency).

Reply:
You can install debug gem the same way you can install `pry-byebug` for older Rubies. It supports Ruby 2.6+. 

Anyway the whole discussion is good sign that this setup is not simple. Experienced developers can use their favourite tools, there is almost nothing specific. And for beginners it would be maybe better to point to some generic Ruby debugging tutorial where basics like puts debugging and real debuggers are explained. :thinking: 

On the other way, if you think @indirect this kind of guide is valuable in here, let's start somewhere. I'm ok with anything that works. I was just about share my experience.

Reply:
Based on this convo, I reorganized the doc slightly and tried to add just enough info to get new maintainers started but would like to link out to other places for more learning. Do we care who/what we link to? Here are some options (not in the doc yet)

- REPL debugging more info: https://st0012.dev/whats-new-in-ruby-3-2-irb
- Prints debugging more info: https://medium.com/@calielrushton/puts-prints-and-more-printing-to-the-console-in-ruby-7830cdaace0a
- Interactive debugging more info: https://github.com/ruby/debug#how-to-use